src/ray/stats/stats.h

```diff
@@ -51,14 +51,23 @@ static void Init(
     RAY_LOG(INFO) << "Disabled stats.";
     return;
   }
+  RAY_LOG(INFO) << "Initialize stats";
 
   // Force to have a singleton exporter.
   static std::shared_ptr<MetricExporterClient> exporter;
   // Default exporter is metrics agent exporter.
   if (exporter_to_use == nullptr) {
     std::shared_ptr<MetricExporterClient> stdout_exporter(new StdoutExporterClient());
-    exporter.reset(new MetricsAgentExporter(stdout_exporter, metrics_agent_port,
-                                            io_service, "127.0.0.1"));
+    try {
+      // To catch exception if core worker fails to init stats when metric agent
```

Comment:
Isn't this method already returned when metric agent is disabled?

Reply:
> Isn't this method already returned when metric agent is disabled?

Try-catch block is added for gtest or some unit tests without metric agent launched, so that an invalid endpoint or port cannot set up a metric exporter client normally.

Reply:
Do we have a better way to cover the case? e.g. check `metrics_agent_port` before creating the exporter. Also, outputting a warning log in totally normal scenarios looks odd to me.